cypher-shell/src/main/java/org/neo4j/shell/commands/Param.java

```diff
@@ -58,21 +62,44 @@ public String getHelp() {
 
     @Override
     public void execute(@Nonnull final String argString) throws CommandException {
-        Matcher alphanumericMatcher = argPattern.matcher(argString);
-        if (alphanumericMatcher.matches()) {
-            variableHolder.set(alphanumericMatcher.group("key"), alphanumericMatcher.group("value"));
-        } else {
-            checkForBackticks(argString);
+        Matcher lambdaMapMatcher = lambdaMapPattern.matcher(argString);
+        if (lambdaMapMatcher.matches()) {
+            throw new CommandException(AnsiFormattedText.from("Incorrect usage.\nusage: ")
+                    .bold().append(COMMAND_NAME).boldOff().append(" ").append(getUsage()));
         }
+        if (!validParameterAssignment(argString)) {
```

Comment:
`validParameterAssignment` looks like a pure validation method but it modifies the variable holder. I'd suggest renaming it to smth like `assignParameterIfPossible`